### PR TITLE
Modularize customization of time series plots 

### LINKED ISSUE #2417: Modularize customization of time series plots
**Is your feature request related to a problem? Please describe.**
In time series plotting module, lot of plots are customized at the end - template, fig size, etc. Since the same code is repeated in all these plots, maybe this could be modularized and reused.

```python
with fig.batch_update():
    template = _resolve_dict_keys(
        dict_=fig_kwargs, key="template", defaults=fig_defaults
    )
    fig.update_layout(title=title, showlegend=False, template=template)

    fig = _update_fig_dimensions(
        fig=fig, fig_kwargs=fig_kwargs, fig_defaults=fig_defaults
    )
```

### PR DESCRIPTION
**Pull request by ryankarlos**
# Related Issue or bug

Closes #2417

# Describe the changes you've made

abstract out redundant code into util functions for customising plots which could be reused in multiple plotting methods/functions

# Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.
- [ ] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Comment by ryankarlos** [comment]
@ngupta23  Will add docstring and test for new util which customises plot.  I currently moved this to `pycaret/internal/plots/utils/time_series.py` but let me know if you prefer it moved to another module.

**Comment by ngupta23** [comment]
@ryankarlos Can you check why the tests are failing? Thanks!

**Comment by ryankarlos** [comment]
please see comment in https://github.com/pycaret/pycaret/pull/3010#issuecomment-1265451890

File: pycaret/internal/plots/time_series.py [hunk 1]
```diff
@@ -270,15 +271,7 @@ def plot_series(
...
-    with fig.batch_update():
-        template = _resolve_dict_keys(
-            dict_=fig_kwargs, key="template", defaults=fig_defaults
-        )
-        fig.update_layout(title=title, showlegend=True, template=template)
-
-        fig = _update_fig_dimensions(
-            fig=fig, fig_kwargs=fig_kwargs, fig_defaults=fig_defaults
-        )
+    fig = _plot_fig_update(fig, title, fig_defaults, fig_kwargs)
```
Comment (ngupta23) [inline pycaret/internal/plots/time_series.py:274]: Should `show_legend` be True here?

**Review by ngupta23 — APPROVED** [review]
Looks good overall. I think there may be a small change needed which I have noted. Please have a look. Thanks!

**Comment by ngupta23** [comment]
> There. are few other cases like below which also update x, y axes and traces , and are duplicated in two to three other plotting methods.

I do not think that those are that pervasive, so it is ok to leave it as is for now.

**Comment by ryankarlos** [comment]
Thanks @ngupta23  pushed new changes and merged master

**Pull request merged**